querybook/webapp/components/DataDocTemplateButton/DataDocTemplateVarForm.tsx

```diff
@@ -44,122 +47,149 @@ const templatedVarSchema = Yup.object().shape({
     ),
 });
 
-const defaultTemplatedVariablesValue = { '': '' };
+/**
+ * This interface is used for drag and drop purposes
+ */
+interface IDataDocMetaVariableWithId extends IDataDocMetaVariable {
+    id: string;
+}
+const templatedVarUniqueIdPrefix = '_tvar';
+
+const defaultTemplatedVariables: IDataDocMetaVariableWithId[] = [
+    {
+        name: '',
+        value: '',
+        type: 'string',
+        id: uniqueId(templatedVarUniqueIdPrefix),
```

Comment:
this means the name could be not unique? Shall we update the Yup and also the backend validation to not allow duplicate names?

Reply:
it could be not unique while the user is writing the form